addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -2215,18 +2288,71 @@ const Many2oneUserValueWidget = SelectUserValueWidget.extend({
         this.searchMore.classList.add('o_we_m2o_search_more');
         this.searchMore.textContent = _t("Search more...");
         this.searchMore.title = _t("Search to show more records");
+
+        if (this.options.createMethod) {
+            this.createInput = new InputUserValueWidget(this, undefined, {
+                classes: ['o_we_large_input'],
+                dataAttributes: { noPreview: 'true' },
+            }, this.$target);
+            this.createButton = new ButtonUserValueWidget(this, undefined, {
+                classes: ['flex-grow-0'],
+                dataAttributes: {
+                    noPreview: 'true',
+                    [this.options.createMethod]: '', // Value through getValue.
+                },
+                childNodes: [document.createTextNode(_t("Create"))],
+            }, this.$target);
+            // Override isActive so it doesn't show up in toggler
+            this.createButton.isActive = () => false;
+
+            // Override button _onUserValueChange so that it only happens if
+            // the input has a value.
+            const prevOnUserValueChange = this.createButton._onUserValueChange;
+            const self = this;
+            this.createButton._onUserValueChange = function () {
+                if (self.createInput._value) {
+                    prevOnUserValueChange.apply(this, arguments);
+                }
+            };
```

Comment:
Is there no other way than in-place monkey-patching? Can't you just ignore the empty values the create stuff receives?